TESTING.md

```diff
@@ -0,0 +1,137 @@
+
+1. Environment setup should contain `test-project/` described in the [Test Setup](#test-setup) section.
+
+2. Run the following Docker command in the `test-project/` directory:
+
+```bash
+docker run --rm -v $PWD:/workdir --workdir=/workdir --entrypoint make python:3 help
+```
+
+3. This should result in the creation of both `.tardigrade-ci` and `tardigrade-ci/`.
+
+4. Run the following command to remove the `tardigrade-ci` subdirectory and bootstrap Makefile `.tardigrade-ci`:
+
+```bash
+docker run --rm -v $PWD:/workdir --workdir=/workdir --entrypoint make python:3 clean
+```
+
+5. Go to parent directory of `test-project/` and remove the directory to delete testing environment:
+
+```bash
+cd .. && rm -rf test-project/
+```
+
+### Via Docker (tardigrade-ci image)
+
+Running a tardigrade-ci container will use the version of `tardigrade-ci` present in the container, and so will create only the `.tardigrade-ci` file in the `test-project/` directory.
+
+1. Environment setup should contain `test-project/` described in the [Test Setup](#test-setup) section.
+
+2. In the `test-project/` directory, run the following command:
+
+```bash
+docker run --rm -v $PWD:/workdir --workdir=/workdir --entrypoint make plus3it/tardigrade-ci:latest help
+```
+
+3. This should result in only the `.tardigrade-ci` file being created.
+
+4. Run the following command to remove the `tardigrade-ci` subdirectory and bootstrap Makefile `.tardigrade-ci`:
+
+```bash
+docker run --rm -v $PWD:/workdir --workdir=/workdir --entrypoint make plus3it/tardigrade-ci:latest clean
+```
+
+5. Go to parent directory of `test-project/` and remove the directory to delete testing environment:
+
+```bash
+cd .. && rm -rf test-project/
+```
+
+### Via Makefile (`tardigrade-ci/` directory does not exist in root)
+
+This option uses `make` to invoke the targets directly in your shell, instead of within the container environment. You may invoke any `make` target directly, e.g. `make ec/lint`, but be aware if you do so, the make target will attempt to install the tools it requires to your system.
+
+1. Environment setup should contain `test-project/` described in the [Test Setup](#test-setup) section.
+
+2. In the `test-project/` directory, run the following command:
+
+```bash
+make help
+```
+
+3. This should result in the creation of both `.tardigrade-ci` and `tardigrade-ci/`.
+
+4. Run the following command to clean up the tardigrade-ci files:
+
+```bash
+make clean
+```
+
+5. Go to parent directory of `test-project/` and remove the directory to delete testing environment:
+
+```bash
+cd .. && rm -rf test-project/
+```
+
+### Via Makefile (`tardigrade-ci/` directory exists in root)
+When the bootstrap Makefile detects the `tardigrade-ci/` in the root directory, it will include the tardigrade-ci Makefile from the user-managed clone into the current working project.
```

Comment:
```suggestion
When the bootstrap Makefile detects the directory `tardigrade-ci/` in a parent of the current project directory, it will "include" that tardigrade-ci Makefile, instead of re-cloning `tardigrade-ci/`.
```